vassal-app/src/main/java/VASSAL/counters/GlobalCommandVisitor.java

```diff
@@ -39,6 +41,9 @@ public class GlobalCommandVisitor implements DeckVisitor {
   private int selectFromDeck;
   private GlobalCommand globalCommand;
 
+  // Keep track of the ID's of all pieces processed by this GKC.
+  private final Map<String, GamePiece> seen = new HashMap<>();
```

Comment:
Shouldn't this be Set and HashSet if you don't care about the values?

Reply:
Indeed. Update pushed.